BranchMetrics/ios-branch-deep-linking-attribution
# SDK-2182 port layout changes for tuist spm

## Reference
SDK-2182
This PR ports over changes that were already released for Tuist SPM on the spm repo. Also includes additional changes to fix integration with cocoapods and carthage.

## Summary
Source code layout has been changed to allow non-Xcode SPM to integrate the Branch SDK. This unblocks tuist users.
Updated cocoapods and carthage builds to account for the new layout.

Also includes a bugfix for an earlier unit test changes around BNCURLFilter, this has not shipped yet.

## Motivation
Fix compatibility with SPM when not building from within Xcode.

## Type Of Change

## Testing Instructions
Confirm SPM can install and build the Branch SDK.
Confirm Cocoapods can install and build the Branch SDK.
Confirm Carthage can install and build the Branch SDK.
Confirm manually built xcframeworks are valid.
Confirm Branch skiplist version is properly stored.









cc @BranchMetrics/saas-sdk-devs for visibility.
